### PR TITLE
Integration Tests for Binaries

### PR DESCRIPTION
**Pull request by reiner-dolp**
- [x] drop useless assert_cmd crate
- [x] remove noise from diff output
- [x] choose diff style dynamically (idea by @problame). Force diff style using env var `DIFF_USING`.
- [x] add env var `UPDATE_REFERENCES` that automatically overwrites reference files with tentatives (wish by @joshuabach)
- [x] always generate tentatives (wish by @flip1995)
- [x] implement binary tests (gen with compiler, assert stderr/stdout/exitcode of binary itself)

### CODE REVIEW

File: compiler-cli/tests/integration.rs [hunk 1]
```diff
@@ -129,90 +134,232 @@ struct TestFiles {
...
+
+/// Build and get the main binary
+fn project_binary(subproject: Option<&'static str>) -> PathBuf {
+    let mut cache = BIN_PATH_CACHE.lock().unwrap();
+
+    if let Some(path) = cache.get(&subproject) {
+        return path.clone();
     }
+
+    let mut cmd = Command::new("cargo");
```
Comment (problame) [inline compiler-cli/tests/integration.rs]: Probably want `std::env::car("CARGO")`?

Comment (reiner-dolp) [reply compiler-cli/tests/integration.rs]: cool thanks!

**Comment by reiner-dolp** [comment]
The runtime fails to link: https://travis-ci.org/comprakt/comprakt/jobs/459363337#L680
I don't know why yet.

**Comment by reiner-dolp** [comment]
blocked on #145.

**Comment by reiner-dolp** [comment]
@problame please review b6dabd8 and 09910fd . They are most likely within your domain of knowledge.

**Comment by problame** [comment]
Those LGTM

**Comment by reiner-dolp** [comment]
bors try

**Comment by flip1995** [comment]
This should work now. Please clean up the commits a bit (especially removing 8ab72bf) and then this is good to go

bors try

**Comment by reiner-dolp** [comment]
I would like to keep 8ab72bf since CC is an active part of the program now (called in tests and the actual compract binary).

**Comment by flip1995** [comment]
Ok, but lets already emit it in the install phase.

**Comment by flip1995** [comment]
bors try

**Pull request merged**